Sources/Validation/datetime.swift

```
@@ -0,0 +1,46 @@
+import Foundation
+
+func validateDateTime(_ context: Context, _ value: Any) -> AnySequence<ValidationError> {
+  if let date = value as? String {
+    if let regularExpression = try? NSRegularExpression(pattern: "^\\d{4}-\\d{2}-\\d{2}T\\d{2}:\\d{2}:\\d{2}", options: .caseInsensitive) {
```

Can you please explain the reasoning on why to use both a regex and then a date formatter to validate the input, are there particular inputs which would both pass one check while failing the other?

It would be good to see some test coverage which demonstrates inputs that the date formatter below would incorrectly validate? A signal that this regex is working correctly. Especially as an invalid regex here would silently falldown to the logic below.

---

> are there particular inputs which would both pass one check while failing the other?

Yes, the Swift date formatters are lenient with certain date formats and will parse e.g. `2002-01-1` without issue. This is rather a brute-force workaround, however it does enforce the requirement